meta/rb_map.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2018 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Robert Y. Lewis
+
+Additional operations on native rb_maps and rb_sets.
+-/
+
+namespace native
+namespace rb_set 
+
+meta def filter {key} (s : rb_set key) (P : key → bool) : rb_set key :=
+s.fold s (λ a m, if P a then m else m.erase a)
+
+meta def union {key} (s t : rb_set key) : rb_set key :=
+s.fold t (λ a t, t.insert a)
+
+end rb_set 
+
+namespace rb_map 
+
+meta def ifind {α β} [inhabited β] (m : rb_map α β) (a : α) : β :=
+match m.find a with 
+| some a := a 
+| none := default β
+end 
+
+meta def zfind {α β} [has_zero β] (m : rb_map α β) (a : α) : β :=
+match m.find a with 
+| some v := v 
+| none := 0
+end

```

Comment:
This is `(m.find a).get_or_else 0`

Reply:
Okay. This and `ifind` are useful compositions, particularly `ifind`. I've used both in other projects besides this. Do you think they don't belong in the library? They have to be in the `rb_map` namespace to work, so I can't just define and use them locally.

Reply:
I'm not saying the definitions are bad, I'm just pointing out that there is a simpler way to write the definition. Feel free to define `ifind` and such if you find it useful